src/Servers/Kestrel/Core/src/CoreStrings.resx

```diff
@@ -307,7 +307,7 @@
     <value>IHttpUpgradeFeature.UpgradeAsync was already called and can only be called once per connection.</value>
   </data>
   <data name="BadRequest_RequestBodyTooLarge" xml:space="preserve">
-    <value>Request body too large.</value>
+    <value>Request body too large. Default value is 30,000,000 bytes, which is approximately 28.6MB. </value>
```

Comment:
We shouldn't hard code this in the exception. We should pass the actual limit here, not a hardcoded limit.

Reply:
Ohh make sense. I will get that fixed!

Reply:
But the problem is how could I get to `_maxRequestBodySize` (or its public property) which is present here https://github.com/dotnet/aspnetcore/blob/4cc933927fb668a67e1ce6f2e368edb45af164b4/src/Servers/Kestrel/Core/src/KestrelServerLimits.cs#L31

in static class here 
https://github.com/dotnet/aspnetcore/blob/4cc933927fb668a67e1ce6f2e368edb45af164b4/src/Servers/Kestrel/Core/src/KestrelBadHttpRequestException.cs#L68

I tried to find

Reply:
An example https://github.com/dotnet/aspnetcore/blob/7a20550cc61bd7661836d69f071919e6850e7867/src/Servers/Kestrel/Core/src/CoreStrings.resx#L211. Plus you need to re-run a command on the command line (remind me @javiercn) to generate a new method (it will be a method because it has arguments)

Reply:
Oh, but we need to pass `requestBufferSize` with `string.Format` correct? Or not?

Reply:
They have access to `KestralOptions` here from where they are passing those two values!
https://github.com/dotnet/aspnetcore/blob/ea72e31a4dc3b332887533a0d7b0dd5917f3782e/src/Servers/Kestrel/Core/src/Internal/KestrelServerImpl.cs#L367-L381

Reply:
@MCCshreyas Everywhere `KestrelBadHttpRequestException.Throw(RequestRejectionReason.RequestBodyTooLarge)` is called, the `MaxRequestBodySize` is accessible. Otherwise the code wouldn't know whether or not to throw.

There's both `Throw(RequestRejectionReason)` and `Throw(RequestRejectionReason, string)`. You can move `RequestRejectionReason.RequestBodyTooLarge` to [the overload that takes `string detail`](https://github.com/dotnet/aspnetcore/blob/c925f99cddac0df90ed0bc4a07ecda6b054a0b02/src/Servers/Kestrel/Core/src/KestrelBadHttpRequestException.cs#L100) and pass the `MaxRequestBodySize` as a string.

Here are the call sites for `KestrelBadHttpRequestException.Throw(RequestRejectionReason.RequestBodyTooLarge)` if you still want to take a crack at fixing this:

- https://github.com/dotnet/aspnetcore/blob/c925f99cddac0df90ed0bc4a07ecda6b054a0b02/src/Servers/Kestrel/Core/src/Internal/Http/MessageBody.cs#L178
- https://github.com/dotnet/aspnetcore/blob/c925f99cddac0df90ed0bc4a07ecda6b054a0b02/src/Servers/Kestrel/Core/src/Internal/Http/Http1ContentLengthMessageBody.cs#L240
- https://github.com/dotnet/aspnetcore/blob/c925f99cddac0df90ed0bc4a07ecda6b054a0b02/src/Servers/Kestrel/Core/src/Internal/Http2/Http2MessageBody.cs#L29
- https://github.com/dotnet/aspnetcore/blob/c925f99cddac0df90ed0bc4a07ecda6b054a0b02/src/Servers/Kestrel/Core/src/Internal/Http3/Http3MessageBody.cs#L29